app/cdap/components/ConnectionList/index.tsx

```diff
@@ -198,76 +280,138 @@ export default function ConnectionList() {
     );
   };
 
+  useEffect(() => {
+    setTabSize(dataForTabs.length);
+  }, [dataForTabs.length]);
+
   let headerContent;
 
   return (
     <Box data-testid="data-sets-parent" className={classes.connectionsListContainer}>
       <SubHeader selectedConnection={dataForTabs[0]?.selectedTab} />
+
       {dataForTabs &&
       Array.isArray(dataForTabs) &&
       dataForTabs.length &&
       dataForTabs[0]?.data?.length > 0 ? (
-        <SelectDatasetWrapper>
-          {dataForTabs.map((each, index) => {
-            const connectionIdRequired =
-              each.data.length && each.data.filter((el) => el.connectionId);
-            if (connectionIdRequired.length) {
-              connectionId = connectionIdRequired[0].connectionId;
-            }
-            if (index === 0) {
-              headerContent = headerForLevelZero();
-            } else {
-              headerContent =
-                refs.current[index]?.offsetWidth < refs.current[index]?.scrollWidth ? (
-                  <CustomTooltip title={dataForTabs[index - 1].selectedTab} arrow>
-                    <Box className={classes.beforeSearchIconClickDisplay}>
-                      <Typography
-                        variant="body2"
-                        className={classes.headerLabel}
-                        ref={(element) => {
-                          refs.current[index] = element;
-                        }}
+        <Box className={classes.connectionsWithInfo}>
+          <SelectDatasetWrapper>
+            {filteredData &&
+              Array.isArray(filteredData) &&
+              filteredData?.map((eachFilteredData, index) => {
+                if (eachFilteredData.data.filter((el) => el.connectionId).length) {
+                  connectionId = eachFilteredData.data.filter((el) => el.connectionId)[0]
+                    .connectionId;
+                }
+                if (index === 0) {
+                  headerContent = headerForLevelZero();
+                } else {
+                  headerContent = (
```

Comment:
we can separate the headerContent component as a separate component, sub component

Reply:
Done